src/auth/awsrequestsigner.ts

```diff
@@ -187,9 +191,17 @@ async function getSigningKey(
  * Generates the authentication header map needed for generating the AWS
  * Signature Version 4 signed request.
  *
- * @param accessKeyId The AWS access key ID.
- * @param secretAccessKey The AWS secret access kye.
- * @param token The AWS token.
+ * @param crypto The crypto instance used to facilitate cryptographic
+ *   operations.
+ * @param host The AWS service URL hostname.
+ * @param canonicalUri The AWS service URL path name.
+ * @param canonicalQuerystring The AWS service URL query string.
+ * @param method The HTTP method used to call this API.
+ * @param region The AWS region.
+ * @param securityCredentials The AWS security credentials.
+ * @param requestPayload The optional request payload if available.
+ * @param additionalAmzHeaders The optional additional headers needed for the
+ *   requested AWS API.
```

Comment:
I know this already shipped and isn't part of the review, but I really advise to use a 1-3 parameters in most cases with JavaScript functions.  For consumers, there's often no typing or tooling (for non-TypeScript users), so taking more than a few parameters can be very hard.  It also significantly impacts code refactorability in the future.  I would really prefer we take a single "GenerateAuthHeaderMapOptions" parameter defined as an interface, that can change and grow over time without worrying about positional parameters.

Reply:
Thanks for the review and feedback @JustinBeckwith. I agree with you. I made the requested change. This specific function is internal only (it is not exported). So it should not be used externally, but regardless, I updated it as this could grow in the future. Using an object makes it definitely more manageable.